pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -831,26 +830,39 @@ func (og *operationGenerator) GenerateUnmountDeviceFunc(
 	}
 
 	unmountDeviceFunc := func() (error, error) {
-		//deviceMountPath := deviceToDetach.DeviceMountPath
 		deviceMountPath, err :=
 			volumeDeviceMounter.GetDeviceMountPath(deviceToDetach.VolumeSpec)
 		if err != nil {
-			// On failure, return error. Caller will log and retry.
-			return deviceToDetach.GenerateError("GetDeviceMountPath failed", err)
+			if !goerrors.Is(err, os.ErrNotExist) {
+				// On failure other than not exist, return error. Caller will log and retry.
+				return deviceToDetach.GenerateError("GetDeviceMountPath failed", err)
+			}
+			// If the mount path could not be found, don't fail the unmount, but instead just use the value from
+			// deviceToDetach.DeviceMountPath and proceed so that the device can be marked as unmounted
+			deviceMountPath = deviceToDetach.DeviceMountPath
+			klog.Warningf(deviceToDetach.GenerateMsg(fmt.Sprintf(
+				"GetDeviceMountPath failed, but unmount operation will continue (using deviceMountPath=%s): %v", deviceMountPath, err), ""))
 		}
 		refs, err := deviceMountableVolumePlugin.GetDeviceMountRefs(deviceMountPath)
 
 		if err != nil || util.HasMountRefs(deviceMountPath, refs) {
 			if err == nil {
 				err = fmt.Errorf("The device mount path %q is still mounted by other references %v", deviceMountPath, refs)
 			}
-			return deviceToDetach.GenerateError("GetDeviceMountRefs check failed", err)
+			if !goerrors.Is(err, os.ErrNotExist) {
+				// On failure other than not exist, return error. Caller will log and retry.
+				return deviceToDetach.GenerateError("GetDeviceMountRefs failed", err)
+			}
+			klog.Warningf(deviceToDetach.GenerateMsg(fmt.Sprintf("GetDeviceMountRefs failed, but unmount operation will continue: %v", err), ""))
 		}
 		// Execute unmount
 		unmountDeviceErr := volumeDeviceUmounter.UnmountDevice(deviceMountPath)
 		if unmountDeviceErr != nil {
-			// On failure, return error. Caller will log and retry.
-			return deviceToDetach.GenerateError("UnmountDevice failed", unmountDeviceErr)
+			if !goerrors.Is(unmountDeviceErr, os.ErrNotExist) {
```

Comment:
Looking at the local volume plugin, it returns `nil` when `deviceMountPath` does not exist. Is this code necessary?

Reply:
Maybe it is not needed for the local volume plugin, but I was thinking that another plugin might return this error, so I want to make sure to handle it, regardless of which plugin is used.

Reply:
I also don't want to rely on internal knowledge about the local volume plugin.
For example, someday the local volume plugin might be changed internally and whoever changes it would not be aware of this assumption being made in operation_generator.go about whether the not exists error is propagated or suppressed.

Reply:
this changed the semantic of this interface, we should be careful. it's possible that the plugin returns `ErrNotExist` because some file it depends on internally does not exist.
how about fixing the plugin instead if some plugin does not return nil when the device mount path does not exist.